### PR TITLE
docs(howto): upload a package with resources

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

-----

Requires #142, #143, #144

### CODE REVIEW

File: docs/howto/upload-package-with-resources.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
+.. _howto-upload_package_with_resources:
+
+Uploading a package with resources
```
[the remark this answers is not in the document]
Comment (lengau) [reply docs/howto/upload-package-with-resources.rst]: 😆 that's what I had initially and then decided it was too long. Changed back!

File: docs/howto/upload-package-with-resources.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+==================================
+
+One of the most common workflows when communicating with the store API is to
+release a package. This guide explains how to upload a package and an associated
+resource and then release that. We will use a charm as an example.
```
Comment (tigarmo) [inline docs/howto/upload-package-with-resources.rst:8]: release those/them?

File: docs/howto/upload-package-with-resources.rst [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    :end-before: [docs:release-end]
+    :dedent: 4
+
+
+Below is a full file containing an executable version of the script in this
+guide.
+
+.. collapse:: Full example
+
+    .. literalinclude:: code/upload-package-with-resources/upload_package.py
```
Comment (tigarmo) [inline docs/howto/upload-package-with-resources.rst:94]: the only downside is that this will contain the `# [docs:x]` markers right?

**Review by tigarmo — APPROVED** [review]
very nice!

File: docs/howto/upload-package-with-resources.rst (hunk 3 above)
Comment (lengau) [reply docs/howto/upload-package-with-resources.rst:94]: That's the only one I know of.

I would've preferred to make this a link to a download of the file (which wouldn't have solved that downside), but I couldn't manage to get it to work nicely in sphinx (the only thing I could come up with was linking to the file on GitHub).

**Pull request merged**